### PR TITLE
Add AGENTS files

### PR DESCRIPTION
**Pull request by dgarros**
Add fairly minimalist `AGENTS.md` files in a 
- root
- backend
- frontend
- docs

## Summary by CodeRabbit

* **Documentation**
  * Added a centralized, repo-wide contributor guide covering project overview, file structure, commands, coding/testing conventions, writing style, and contributor boundaries.
  * Added targeted guidance for backend, frontend, and docs authors; replaced an older guidance file with a pointer to the central guide.
* **Chores**
  * Expanded spelling exceptions to include additional accepted terms.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: frontend/app/AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+### Import Order (Biome enforced)
+
+```typescript
+import { useState } from "react";           // React
+import { useQuery } from "@apollo/client";  // External packages
+import { useConfig } from "@/config";       // Internal aliases
+import { NodeCard } from "@/shared/...";    // Shared
+import { useNodeData } from "@/entities/..."; // Entities
+import "./styles.css";                      // Local
+```
```
Comment (bilalabbad) [inline frontend/app/AGENTS.md]: A big parts of this are inaccurate and don’t align with our project conventions. Feel free to reduce it to the bare minimum or delete the file altogether. I’ll be writing AGENTS.md today anyway.

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+cd frontend/app && npm run build      # Build frontend
+cd docs && npm run build              # Build documentation
+```
+
+## Generated Files (Do Not Edit)
+
+- `backend/infrahub/core/schema/generated/` – Schema definitions
+- `backend/infrahub/core/protocols.py` – Protocol definitions
+- `frontend/app/src/shared/api/graphql/generated/` – GraphQL types
+- `frontend/app/src/shared/api/rest/types.generated.ts` – REST types
```
Comment (ogenstad) [inline AGENTS.md:66]: Add ./schema/schema.graphql and ./schema/openapi.json

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+- **Commit messages:** Conventional commits; include issue references
+- **Changelog:** Add fragment to `changelog/` using Towncrier format
+
+## Boundaries
+
+### Always Do
+
+- Run formatters before committing (`uv run invoke format`, `npm run biome:fix`)
+- Write tests for new functionality
+- Use type hints (Python) and TypeScript types (frontend)
```
Comment (ogenstad) [inline AGENTS.md]: The meaning here looks a bit weird with Python and frontend within parentheses. 

Would be clearer with something like:
Use type hints for Python (backend) and TypeScript types (frontend)

File: backend/AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+### Async-First
+
+```python
+# ✅ Good
+async def get_node(db: InfrahubDatabase, node_id: str) -> Node:
+    query = await NodeGetQuery.init(db=db, node_id=node_id)
+    await query.execute(db=db)
+    return query.get_node()
+
+# ❌ Bad - blocks event loop
```
Comment (ogenstad) [inline backend/AGENTS.md]: Bad - blocks event loop, no type hints

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+uv run invoke backend.test-integration # Backend integration tests
+cd frontend/app && npm run test       # Frontend unit tests
+cd frontend/app && npm run test:e2e   # Frontend E2E tests
+```
+
+### Linting & Formatting
+
+```bash
+uv run invoke format                  # Format all Python code
+uv run invoke lint                    # Lint all Python code
```
Comment (ogenstad) [inline AGENTS.md:48]: I'm not sure here, but to me this would be duplicated info and something that's also included in the backend. I don't know if it hurts to have it here or if it just creates more info in the context. I.e. the Python stuff might not be relevant for work within the frontend and vice versa.

I guess we'll have to test and reiterate on things like this.

File: backend/AGENTS.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+### Pydantic Models
+
+```python
+# ✅ Good
+from pydantic import BaseModel, Field
+
+class BranchCreateInput(BaseModel):
+    name: str = Field(..., min_length=1, max_length=250)
+    description: str | None = None
```
Comment (ogenstad) [inline backend/AGENTS.md]: I'd suggest making both of these as Field objects and providing a description parameter with some meaningful text.

File: backend/AGENTS.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    name: str = "my_query"
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs) -> None:
+        self.params["node_id"] = kwargs["node_id"]
+        self.add_to_query("MATCH (n:Node {uuid: $node_id}) RETURN n")
+```
+
+## Testing
+
+- Unit tests: no external dependencies, mock everything
```
Comment (ogenstad) [inline backend/AGENTS.md]: I'd like to get rid of the "mock" part and instead focus on having APIs that can use a protocol, interface or adapter. With mocks we can end up with things that doesn't reflect the real world and if something external to the function changes those tests can still pass. I'd like to get rid of all of the mocks throughout the tests.

File: docs/AGENTS.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+  - `topics/` – Explanations (understanding-oriented)
+  - `reference/` – API/configuration reference
+  - `tutorials/` – Learning tutorials
+  - `media/` – Images and screenshots
+- `sidebars.ts` – Navigation configuration
+
+## Commands
+
+```bash
+npm run start          # Dev server with hot reload
```
Comment (ogenstad) [inline docs/AGENTS.md:22]: As this is a blocking call that doesn't return the prompt, can Claude do that and understand to run it as a background process? (I don't currently know)

**Review by ogenstad — APPROVED** [review]
Added some suggestions.

File: docs/AGENTS.md (hunk 8 above)
Comment (dgarros) [reply docs/AGENTS.md:22]: Not sure, will need to test that
